### PR TITLE
feat: use monospace font in capability names and values so whitespace is obvious

### LINKED ISSUE #188: feature request: use monospace font in capability names and values so whitespace is obvious
## Appium or Appium Desktop?

You are reporting an issue at the Appium Desktop repository. Appium Desktop is a wrapper around Appium. If you are having trouble running tests, it is much more likely that the problem you are encountering is not a problem with Appium Desktop but with Appium. For that reason we require you to have tried your testcase with Appium CLI before reporting issues. Check the checkbox below to confirm that you have proven your issue does _not_ reproduce on Appium itself:

- [x] I have verified that my issue does not occur with Appium and should be investigated as an Appium Desktop issue

## The problem

I use a lot of time to get the reason of the error:
image
finally, I find the beginning whitespace!
image

_So I advise to auto remove the whitespace from the beginning and end of the string in Capability value_

## Environment

- I am running Appium Desktop version _1.22.0_.
- I am on (pick one):
    - [ ] Mac
    - [x] Windows
    - [ ] Linux

### PR DESCRIPTION
**Pull request by emaan-c**
This PR Closes #188 

Changed the font in the capabilities text boxes to monospace so that whitespaces are more apparent!

image

### CODE REVIEW

**Comment by jlipps** [comment]
the trick would be to figure out which characters to consider "invalid" or "dangerous". any ideas there?

**Comment by mykola-mokhnach** [comment]
I would make it simple.
We may simply make whitespaces wisible by for example switching the font as described in https://stackoverflow.com/questions/7900655/visualize-whitespaces-using-css or via css  https://discourse.wicg.io/t/visualizing-whitespace-characters/1394

**Comment by jlipps** [comment]
if there's an easy way to incorporate a font that will do this, that seems like a good idea

**Comment by emaan-c** [comment]
I used Jonathan's suggestion in terms of warning the user if they are using any non-ASCII characters when the user is entering their capabilities:

image

**Comment by mykola-mokhnach** [comment]
> I used Jonathan's suggestion in terms of warning the user if they are using any non-ASCII characters when the user is entering their capabilities:
> 
> image

I'm not quite sure such limitation makes sense for values, but keys

**Comment by emaan-c** [comment]
> > I used Jonathan's suggestion in terms of warning the user if they are using any non-ASCII characters when the user is entering their capabilities:
> > image
> 
> I'm not quite sure such limitation makes sense for values, but keys

Shouldn't this limitation be for values as well, since appium won't be able to run any set of capabilities that have any non-ASCII characters in it?

**Comment by jlipps** [comment]
> Shouldn't this limitation be for values as well, since appium won't be able to run any set of capabilities that have any non-ASCII characters in it?

I think my suggestion of non-ascii is too aggressive. and @mykola-mokhnach is right that we care less about values, since maybe this is like some user generated thing.

I kinda think we're overthinking this somewhat. Is there a simple way to just warn if there is leading or trailing whitespace? that would resolve the concrete issue that has been faced in the past. I'm not sure if other scenarios are likely to actually be problematic. what do you think @mykola-mokhnach ?

**Comment by emaan-c** [comment]
> I think my suggestion of non-ascii is too aggressive. and @mykola-mokhnach is right that we care less about values, since maybe this is like some user generated thing.
> 
> I kinda think we're overthinking this somewhat. Is there a simple way to just warn if there is leading or trailing whitespace? that would resolve the concrete issue that has been faced in the past. I'm not sure if other scenarios are likely to actually be problematic. what do you think @mykola-mokhnach ?

Sounds great, I made a quick change where the tooltip only warns the user when the text box contains any leading/trailing whitespace!

image

**Comment by jlipps** [comment]
I think this works pretty well! @mykola-mokhnach are you good with this?

**Comment by mykola-mokhnach** [comment]
> I think this works pretty well! @mykola-mokhnach are you good with this?

Yes, it's better now. Could we improve that and be more explicit about when leading/training/both whitespaces are present?

**Comment by emaan-c** [comment]
> @emaan-c can we do that, like update the tooltips to say one of the following
> 
> * `Contains Leading Whitespace`
> * `Contains Trailing Whitespace`
> * `Contains Leading and Trailing Whitespace`
> 
> based on the situation?

Yup, I have made those changes!

image

File: app/renderer/components/Session/CapabilityEditor.js [hunk 1]
```diff
@@ -11,6 +11,16 @@ import { ROW } from '../../../../gui-common/components/AntdTypes';
 
 const {Item: FormItem} = Form;
 const {Option} = Select;
+const whitepaces = /^\s|\s$/;
```
Comment (jlipps) [inline app/renderer/components/Session/CapabilityEditor.js:14]: `whitespaces`

**Review by jlipps — APPROVED** [review]
one typo then we're good to go

**Pull request merged**